airbyte-cdk/python/airbyte_cdk/sources/file_based/file_based_source.py

```diff
@@ -278,3 +279,7 @@ def _validate_and_get_validation_policy(self, stream_config: FileBasedStreamConf
     def _validate_input_schema(self, stream_config: FileBasedStreamConfig) -> None:
         if stream_config.schemaless and stream_config.input_schema:
             raise ValidationError("`input_schema` and `schemaless` options cannot both be set", model=FileBasedStreamConfig)
+
+    def discover(self, logger: logging.Logger, config: Mapping[str, Any]) -> AirbyteCatalog:
+        streams = [stream.as_airbyte_stream() for stream in self.streams(config=config) if stream.get_json_schema()]
```

Comment:
should we use availability strategy as we do it in `check_connection`?

Reply:
It is a good suggestion, but using availability strategy involves additional listing of matching files and additional requests to the source, whereas `get_json_schema()` uses cache and will probably work faster. Does that make sense?

Reply:
But if the stream is empty, get_json_schema will raise an error, and the behavior will be the same as it is now.
Or I miss something?

Reply:
After this update, the `get_json_schema` method will not raise an error on empty streams. Instead, it will return an empty `dict` in order to avoid unwanted errors during discover